packages/console/src/ConsoleObjectsMenu.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { ReactElement, useMemo, useRef, useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  Button,
+  DropdownActions,
+  DropdownMenu,
+  SearchInput,
+} from '@deephaven/components';
+import { vsGraph, vsTriangleDown } from '@deephaven/icons';
+import type { dh as DhType } from '@deephaven/jsapi-types';
+import { ObjectIcon } from './common';
+
+interface ConsoleObjectsMenuProps {
+  openObject: (object: DhType.ide.VariableDefinition) => void;
+  objects: readonly DhType.ide.VariableDefinition[];
+}
+
+function ConsoleObjectsMenu({
+  openObject,
+  objects,
+}: ConsoleObjectsMenuProps): ReactElement {
+  const [filterText, setFilterText] = useState('');
+  const searchRef = useRef<SearchInput>(null);
+
+  const actions: DropdownActions = useMemo(() => {
+    if (objects.length === 0) {
+      return [];
+    }
+    const searchActions = {
+      menuElement: (
+        <SearchInput
+          value={filterText}
+          onChange={e => {
+            setFilterText(e.target.value);
+          }}
+          className="console-menu"
+          ref={searchRef}
+        />
+      ),
+    };
+    const filteredObjects = filterText
+      ? objects.filter(
+          ({ title }: { title?: string }) =>
```

Comment:
Don't think you need to specify the type here. The JSAPI types say it's always defined and the source looks to confirm that